modules/apps/commerce/commerce-order-rule-web/src/main/resources/META-INF/resources/cor_entry/type/minimum_amount.jsp

```diff
@@ -26,11 +29,35 @@ COREntryDisplayContext corEntryDisplayContext = (COREntryDisplayContext)request.
 			bodyClasses="flex-fill"
 			title='<%= LanguageUtil.get(request, "configuration") %>'
 		>
-			<aui:input name='<%= "type--settings--" + COREntryConstants.TYPE_MINIMUM_ORDER_AMOUNT_FIELD_CURRENCY_CODE + "--" %>' type="hidden" value="<%= HtmlUtil.escape(corEntryDisplayContext.getDefaultCommerceCurrencyCode()) %>" />
+			<aui:select label="currency" name="<%= selectName %>" required="<%= true %>" title="currency">
+
+				<%
+				for (CommerceCurrency commerceCurrency : corEntryDisplayContext.getCommerceCurrencies()) {
+					String commerceCurrencyCode = commerceCurrency.getCode();
+				%>
+
+					<aui:option label="<%= commerceCurrency.getName(locale) %>" selected="<%= commerceCurrencyCode.equals(corEntryDisplayContext.getCommerceCurrencyCode()) %>" value="<%= commerceCurrencyCode %>" />
 
-			<aui:input label="minimum-order-amount" name='<%= "type--settings--" + COREntryConstants.TYPE_MINIMUM_ORDER_AMOUNT_FIELD_AMOUNT + "--" %>' required="<%= true %>" suffix="<%= HtmlUtil.escape(corEntryDisplayContext.getDefaultCommerceCurrencyCode()) %>" type="text" value="<%= corEntryDisplayContext.getMinimumAmount() %>">
+				<%
+				}
+				%>
+
+			</aui:select>
+
+			<aui:input label="minimum-order-amount" name="<%= inputName %>" required="<%= true %>" suffix="<%= HtmlUtil.escape(corEntryDisplayContext.getCommerceCurrencyCode()) %>" type="text" value="<%= corEntryDisplayContext.getMinimumAmount() %>">
```

Comment:
I think we don't really need to manage the suffix if we have the field before telling us the currency used.
cc @riccardo-alberti